shotover/src/transforms/kafka/sink_cluster/mod.rs

```diff
@@ -2860,3 +2925,50 @@ impl<'a> std::fmt::Display for FormatTopicName<'a> {
         }
     }
 }
+
+/// Partitions the given shotover nodes based on their rack placement.
+/// Returns a tuple of two vectors: (shotover_nodes_in_rack, shotover_nodes_out_of_rack).
+/// If rack is None, all shotover nodes are considered to be in the rack.
+fn partition_shotover_nodes_by_rack<'a>(
+    shotover_nodes: &[&'a ShotoverNode],
+    rack: &Option<&StrBytes>,
+) -> (Vec<&'a ShotoverNode>, Vec<&'a ShotoverNode>) {
```

Comment:
I am very cautious about returning tuples containing values of the same type, since its easy to get it mixed up which value is which.

Can we define a:
```rust
struct ShotoverNodes<'a> {
  in_rack: Vec<&'a ShotoverNode>,
  out_of_rack: Vec<&'a ShotoverNode>
}
```
and return that instead?
It will also mean we can take `ShotoverNodes` as an argument in functions like `select_shotover_node_by_hash`.

Reply:
added this struct